### ISSUE TITLE
SSR: ssr: false and variables in queries

### ISSUE THREAD

**Issue by curio77** [apollostack/apollo-client#1163]
I have a setup using ApolloClient (v0.6.0; v0.7.1 behaves identically) in conjunction with React, Redux, and SSR. With this I noticed that if I have a query decorating a component like this:

```es6
graphql(ItemsQuery, {
  options: ({ root }) => ({
    variables: {
      root: root || null,
      limit: 50,
      offset: null,
    },
  }),
})(Items);
```

with the query being

```graphql
query Items($root: Int, $limit: Int, $offset: Int) {
  items(root:$root limit:$limit offset:$offset) {
    id
    name
  }
}
```

then things work as intended with SSR enabled.

However, adding `ssr: false` to the options gives the following — obviously unfounded — error:

```
Error: The inline argument "root" is expected as a variable but was not provided.
```

I.e., this only happens client-site if SSR is disabled for the query, but not server-side (with SSR enabled).

I'm clueless as to why this might be and suspect a bug in ApolloClient.

**Comment by helfer** [comment]
@curio77 can you reproduce this issue with only Apollo Client (i.e. without using react-apollo
s graphql HOC)? If not, it would be better to file the issue on react-apollo so we can first figure out if it's a bug in react-apollo. Thanks!

**Comment by curio77** [comment]
Drat, completely forgot about `react-apollo` being separate. Will check and update/repost this accordingly.

**Comment by curio77** [comment]
OK, so I've refactored this to do away with `react-apollo`, replacing the decoration by this code in my component's `componentDidMount()`:

```es6
return this.context.apollo.query({
  query: ItemsQuery,
  variables: {
    root: this.props.root || null,
    limit: 50,
    offset: null,
  },
}).then(data => {
  this.setState({ data });
});
```

This yields the same, so I think this *is* an `apollo-client` issue after all?

```
Uncaught (in promise) Error: Network error: The inline argument "root" is expected as a variable but was not provided.
```

**Comment by curio77** [comment]
FWIW, I load my queries via `graphql-tag/loader` but also tried the `gql` template literal rout just now: makes no difference.

**Comment by curio77** [comment]
This seems related to #1183: just as happens there, I've now looked at the chatter between client and GraphQL backend via Wireshark and confirmed that the request goes out including variables and the expected response gets returned, but regardless of this, the error gets thrown and my component gets no data.

**Comment by curio77** [comment]
This issue is resolved for me with v0.8.1.
